### PR TITLE
Handle Water Sampling with AbsoluteFreeEnergy

### PR DESCRIPTION
**Pull request by badisa**
* Low priority bug where we can't run water sampling with InitialState objects constructed from AbsoluteFreeEnergy. User can work around this themselves by unwrapping the summed potential, but thought I would put up a quick fix. 
* Unable to find NonbondedInteractionGroup triggers failure due to wrapping up with a SummedPotential

### CODE REVIEW

File: timemachine/fe/free_energy.py [hunk 1]
```diff
@@ -452,7 +455,19 @@ def prepare_host_edge(self, ff: Forcefield, host_config: HostConfig, lamb: float
         host_bps, host_masses = openmm_deserializer.deserialize_system(host_config.omm_system, cutoff=1.2)
         hgt = topology.HostGuestTopology(host_bps, self.top, host_config.num_water_atoms, ff, host_config.omm_topology)
 
-        final_params, final_potentials = self._get_system_params_and_potentials(ff_params, hgt, lamb)
+        final_params = []
```
Comment (badisa) [inline timemachine/fe/free_energy.py:458]: This is a definitely a cludge, but briefly looking at trying to emulate the behavior of the SingleTopology class seemed like a lot more details to get right than to just unwrap the potentials. This way the final potentials are in the same structure as what is constructed using the `SingleTopology.combine_with_host`.

File: timemachine/fe/free_energy.py [hunk 2]
```diff
@@ -452,11 +455,23 @@ def prepare_host_edge(self, ff: Forcefield, host_config: HostConfig, lamb: float
...
         hgt = topology.HostGuestTopology(host_bps, self.top, host_config.num_water_atoms, ff, host_config.omm_topology)
 
-        final_params, final_potentials = self._get_system_params_and_potentials(ff_params, hgt, lamb)
-        combined_masses = self._combine(ligand_masses, host_masses)
-        return final_potentials, final_params, combined_masses
+        final_params = []
+        final_potentials = []
+        combined_params, combined_potentials = self._get_system_params_and_potentials(ff_params, hgt, lamb)
+        for params, pot in zip(combined_params, combined_potentials):
+            # Unpack the summed potential to be consistent with SingleTopology
```
Comment (jkausrelay) [inline timemachine/fe/free_energy.py:462]: (expand the commnet): and so that downstream code which relies on the potential type works properly

**Review by jkausrelay — APPROVED** [review]

File: timemachine/fe/free_energy.py (hunk 2 above)
Comment (badisa) [reply timemachine/fe/free_energy.py]: Fixed in https://github.com/proteneer/timemachine/pull/1431/commits/e5baf827cb46666226eac8edf162f70445c6dad3

File: timemachine/fe/free_energy.py [hunk 3]
```diff
@@ -452,11 +455,24 @@ def prepare_host_edge(self, ff: Forcefield, host_config: HostConfig, lamb: float
...
+        final_potentials = []
+        combined_params, combined_potentials = self._get_system_params_and_potentials(ff_params, hgt, lamb)
+        for params, pot in zip(combined_params, combined_potentials):
+            # Unpack the summed potential to be consistent with SingleTopology and so
+            # that downstream code which relies on the potential types works properly
+            # TBD: Deboggle and unify the topology classes
+            if isinstance(pot, SummedPotential):
+                for partial_params, sub_pot in zip(pot.params_init, pot.potentials):
+                    final_params.append(partial_params)
+                    final_potentials.append(sub_pot)
```
Comment (maxentile) [inline timemachine/fe/free_energy.py:468]: This is correct for one level of nesting -- perhaps raise error if `isinstance(sub_pot, SummedPotential)` ?

**Review by maxentile — APPROVED** [review]

File: timemachine/fe/free_energy.py (hunk 3 above)
Comment (badisa) [reply timemachine/fe/free_energy.py]: Added in https://github.com/proteneer/timemachine/pull/1431/commits/9d67be72ea9641cbabe4246c9e59d4c9ad8a5914https://github.com/proteneer/timemachine/pull/1431/commits/9d67be72ea9641cbabe4246c9e59d4c9ad8a5914

**Pull request merged**